packages/block-editor/src/components/block-settings-menu-controls/index.js

```diff
@@ -44,7 +46,13 @@ const BlockSettingsMenuControlsSlot = ( {
 	);
 
 	const { canLock } = useBlockLock( selectedClientIds[ 0 ] );
+	const { canRename } = useBlockRename( selectedBlocks[ 0 ] );
 	const showLockButton = selectedClientIds.length === 1 && canLock;
+	const showRenameButton =
+		selectedClientIds.length === 1 &&
+		// Todo confirm whether following conditional is needed anymore.
+		// clientId === selectedClientIds[ 0 ] &&
+		canRename;
```

Comment:
Just checking if this `todo` was for future reference, or to figure out before 6.4 is released?

Reply:
Good catch. I believe we can remove it as now we've moved the code location there's no need to check the menu applies to the current block.